File: jambeth/jAmbeth.Cache/src/de/osthus/ambeth/proxy/CacheEntityFactory.java
package de.osthus.ambeth.proxy;

import de.osthus.ambeth.ioc.IBeanContextAware;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.merge.model.IEntityMetaData;

public class CacheEntityFactory extends EntityFactory
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Override
	protected void postProcessEntity(Object entity, IEntityMetaData metaData)
	{
		if (entity instanceof IBeanContextAware)
		{
			((IBeanContextAware) entity).setBeanContext(beanContext);
		}
		super.postProcessEntity(entity, metaData);
	}
}

File: ambeth/Ambeth.Cache/ambeth/proxy/CacheEntityFactory.cs
using System;
using De.Osthus.Ambeth.Ioc;
using De.Osthus.Ambeth.Log;
using De.Osthus.Ambeth.Merge.Model;

namespace De.Osthus.Ambeth.Proxy
{
    public class CacheEntityFactory : EntityFactory
    {
	    [LogInstance]
	    public new ILogger Log { private get; set; }

	    protected override void PostProcessEntity(Object entity, IEntityMetaData metaData)
	    {
		    if (entity is IBeanContextAware)
		    {
			    ((IBeanContextAware) entity).BeanContext = BeanContext;
		    }
		    base.PostProcessEntity(entity, metaData);
	    }
    }
}
